src/python/todo.py

```
@@ -58,10 +73,21 @@ def add_task(self, task, deadline=None, category=None, notes=None, priority=None
+        if not task or not isinstance(task, str):
+            log.error("Invalid task parameter: task must be a non-empty string")
+            raise ValueError("Task must be a non-empty string")
+
+        try:
+            task_id = self.db.add_task(task, deadline, category, notes, priority)
+            self.refresh_tasks()
+            log.info("Task '%s' added successfully!", task)
+            return task_id
+        except TimeoutError as e:
+            log.error("Timeout while adding task: %s", str(e))
+            raise RuntimeError(f"Connection timeout: {e}")
```

Ruff (0.8.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)